### DISCUSSION TITLE
Prisma internal types not exported

### DISCUSSION THREAD

**Discussion by jgb-solutions** [prisma/prisma#14519] [Q&A] [answered]
I would like to use the internal generated types of the Prisma client in other functions for params, dymanic queries and so on. As an example if I want to build a dynamic where clause, I'd want to create a variable and assign it a type of the where clause generated for that model. For not I can't access those types. Tried `Prisma['{dynamic_type_here}']` but nothing shows up.

**Comment by austincrim** [comment, marked as answer]
Hey @jgb-solutions 👋,

We expose a lot of types on the `Prisma` import itself, is there something missing from there that you need?

```ts
import { Prisma } from '@prisma/client'

const whereInput: Prisma.UserWhereInput
```

**Reply by jgb-solutions** [reply]
@austincrim hey thanks for reaching out. Here's a quick video showing the available types I get. Unfortunately I don't see any where input on it.

https://user-images.githubusercontent.com/8051752/181295133-f581deed-1545-47cf-91d7-93aac0a6dc41.mov

**Reply by jgb-solutions** [reply]
Update: The types are indeed exported from Prisma. I accessed them by using `Prisma.{theType)` after a colon instead of trying to access them on `Prisma` as methods or properties.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
